invenio_app_ils/ui/src/invenio_app_ils/common/components/ESSelector/ESSelectorLoanRequest.js

```diff
@@ -0,0 +1,149 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Form, Header, Modal, Segment } from 'semantic-ui-react';
+import { ESSelector } from './';
+import './ESSelector.scss';
+import { DateTime } from 'luxon';
+import { invenioConfig } from '../../config';
+import { toShortDate, toUTCShortDate } from '../../api/date';
+import { DateRangePicker } from '../DateRangePicker';
+import { isEmpty } from 'lodash';
+
+export const PatronSearchInputContext = React.createContext('false');
+
+export default class ESSelectorLoanRequest extends Component {
+  state = { visible: false };
+
+  constructor(props) {
+    super(props);
+    this.selectorRef = null;
+    this.state = { missingPatron: 'false' };
+    const tomorrow = DateTime.local().plus({ days: 1 });
+    const loanDuration = new DateTime(
+      tomorrow.plus({ days: invenioConfig.circulation.defaultDuration })
+    );
+    this.state = {
+      fromDate: props.defaultStartDate
+        ? props.defaultStartDate
+        : toShortDate(tomorrow),
+      toDate: props.defaultEndDate
+        ? props.defaultEndDate
+        : toShortDate(loanDuration),
+      deliveryMethod: this.deliveryMethods()[1].value,
+    };
+  }
+
+  toggle = () => this.setState({ visible: !this.state.visible });
+
+  save = () => {
+    const { onSave } = this.props;
+    if (isEmpty(this.props.selections)) {
+      this.setState({ missingPatron: 'true' });
+    } else {
+      if (onSave) {
+        onSave(
+          this.props.selections,
+          toUTCShortDate(this.state.fromDate),
+          toUTCShortDate(this.state.toDate),
+          this.deliveryMethods()[1].value
+        );
+      }
+      this.toggle();
+    }
+  };
+
+  handleDateChange = (name, value) => {
+    this.setState({ [name]: value });
+  };
+
+  handleDeliveryMethodChange = (event, object) => {
+    this.setState({ deliveryMethod: object.value });
+  };
+
+  deliveryMethods = () => {
+    return invenioConfig.circulation.deliveryMethods.map(method => {
+      return { key: method, value: method, text: method };
+    });
+  };
+
+  render() {
+    const { title, content, selectorComponent, size } = this.props;
+    const trigger = React.cloneElement(this.props.trigger, {
+      onClick: this.toggle,
+    });
+    const Selector = selectorComponent ? selectorComponent : ESSelector;
+
+    return (
+      <Modal
+        id="es-selector-modal"
+        open={this.state.visible}
+        trigger={trigger}
+        size={size}
+        centered={false}
+        onClose={this.toggle}
+      >
+        <Modal.Header>{title}</Modal.Header>
+        <Modal.Content>
+          <p>{content}</p>
+          <PatronSearchInputContext.Provider value={this.state.missingPatron}>
```

Comment:
I would say maybe that it should be more specific.
`value` -> `inputSearchError` or similar

Does it make sense to disable the confirmation button in the ESSelector too if nothing selected?

@topless is interested in this feature.

Reply:
yeah, you can disable, I was thinking about it, but I chose to have the red outline around the search because it says more explicitly what is wrong, instead of disabling the button